management/server/policy.go

```diff
@@ -0,0 +1,416 @@
+package server
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/netbirdio/netbird/management/server/activity"
+	"github.com/netbirdio/netbird/management/server/status"
+
+	"github.com/open-policy-agent/opa/rego"
+	log "github.com/sirupsen/logrus"
+)
+
+// PolicyUpdateOperationType operation type
+type PolicyUpdateOperationType int
+
+const (
+	// UpdatePolicyName indicates a policy name update operation
+	UpdatePolicyName PolicyUpdateOperationType = iota
+	// UpdatePolicyDescription indicates a policy description update operation
+	UpdatePolicyDescription
+	// UpdatePolicyStatus indicates a policy status update operation
+	UpdatePolicyStatus
+	// UpdatePolicyQuery indicates a policy query update operation
+	UpdatePolicyQuery
+)
+
+// PolicyTrafficActionType action type for the firewall
+type PolicyTrafficActionType string
+
+const (
+	// PolicyTrafficActionAccept indicates that the traffic is accepted
+	PolicyTrafficActionAccept = PolicyTrafficActionType("accept")
+	// PolicyTrafficActionDrop indicates that the traffic is dropped
+	PolicyTrafficActionDrop = PolicyTrafficActionType("drop")
+)
+
+// PolicyUpdateOperation operation object with type and values to be applied
+type PolicyUpdateOperation struct {
+	Type   PolicyUpdateOperationType
+	Values []string
+}
+
+//go:embed rego/default_policy_module.rego
+var defaultPolicyModule string
+
+//go:embed rego/default_policy.rego
+var defaultPolicy string
+
+// PolicyMeta is the metadata of the policy
+type PolicyMeta struct {
+	// Action policy accept or drops packets
+	Action PolicyTrafficActionType `json:"action"`
+
+	// Destinations policy destination groups
+	Destinations []string `json:"destinations"`
+
+	// Sources policy source groups
+	Sources []string `json:"sources"`
+
+	// Port port of the service or range of the ports, and optional protocol (by default TCP)
+	Port string `json:"port"`
+}
+
+// Copy returns a copy of the policy.
+func (pm *PolicyMeta) Copy() *PolicyMeta {
+	return &PolicyMeta{
+		Action:       pm.Action,
+		Destinations: pm.Destinations[:],
+		Sources:      pm.Sources[:],
+		Port:         pm.Port,
+	}
+}
+
+// Policy of the Rego query
+type Policy struct {
+	// ID of the policy
+	ID string
+
+	// Name of the Policy
+	Name string
+
+	// Description of the policy visible in the UI
+	Description string
+
+	// Disabled status of the policy
+	Disabled bool
+
+	// Query of Rego the policy
+	Query string
```

Comment:
As for the initial release, does it make sense to store the default query until we introduce support for custom queries?